### PR TITLE
Add to calendar functionality

### PR DESCRIPTION
**Pull request by dtfreemn**
This implements add to calendar functionality. I already made a task in the task tracker to update it later to store the id in Airtable for better UX.

> NOTE: Abstracted out the Alert as a helper because it is now used in multiple components. Will provide easier maintenance/control over error alerting to have it live in one place and be used across the board.

Pre-populated modal on click of link:
image

Proof of event in calendar:
image

### CODE REVIEW

File: mobile/app/components/Detail/DetailCalendarInfo.tsx [hunk 1]
```diff
@@ -1,29 +1,61 @@
...
+      title: eventName,
+      startDate: startDate.utc().format('YYYY-MM-DDTHH:mm:ss.SSS[Z]'),
+      endDate: endDate.utc().format('YYYY-MM-DDTHH:mm:ss.SSS[Z]'),
+      location: `${eventLocationName} ${eventLocationAddress}`.trim(),
+      notes: eventDescription,
+      allDay: Boolean(!startDate && !endDate),
+    };
+
+    try {
+      return await AddCalendarEvent.presentEventCreatingDialog(addEventConfig);
```
Comment (kevinhylant) [inline mobile/app/components/Detail/DetailCalendarInfo.tsx:53]: is there an option to not present the dialogue & just add it?

At ClassPass we had simply added it to their calendar & shown an alert telling them it had been added successfully.

Maybe we can ask the team next week which option they prefer?

File: mobile/ios/mobile/Info.plist [hunk 2]
```diff
@@ -87,5 +87,9 @@
 		<string>Ubuntu-MediumItalic.ttf</string>
 		<string>Ubuntu-Regular.ttf</string>
 	</array>
+	<key>NSCalendarsUsageDescription</key>
+	<string>The Brave House would like to access your calendar.</string>
+	<key>NSContactsUsageDescription</key>
```
Comment (kevinhylant) [inline mobile/ios/mobile/Info.plist:92]: do we need this about accessing contacts?

**Comment by kevinhylant** [comment]
Other than the contacts bit, this LGTM!

File: mobile/app/components/Detail/DetailCalendarInfo.tsx (hunk 1 above)
Comment (dtfreemn) [reply mobile/app/components/Detail/DetailCalendarInfo.tsx:53]: there's not an option in this library. would have to take the granular approach if we want to bypass the modal.

File: mobile/ios/mobile/Info.plist (hunk 2 above)
Comment (dtfreemn) [reply mobile/ios/mobile/Info.plist:92]: not sure. i can try it without. the docs for the library say to add both, but perhaps the contacts bit is for some functionality that we aren't using.

Comment (dtfreemn) [reply mobile/ios/mobile/Info.plist]: removed.

**Pull request merged**